src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,37 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.Collections;
+import java.util.List;
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @SuppressWarnings("unchecked")
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        if (type == Bulldozer.class) {
+            return (List<T>) new BulldozerProducer().get();
```

Comment:
Consider creating a local variable for the result of the `get()` method to avoid code duplication and make the method easier to maintain.